File: kevoree-library/javase/org.kevoree.library.javase.bluetooth/src/main/scala/org/kevoree/channels/Tester.scala
package org.kevoree.channels

import javax.bluetooth._

/**
 * Created by IntelliJ IDEA.
 * User: duke
 * Date: 23/09/11
 * Time: 14:33
 * To change this template use File | Settings | File Templates.
 */

object Tester extends App {

  var remotes : List[RemoteDevice] = List()


  val localDevice = LocalDevice.getLocalDevice
  println("Address: " + localDevice.getBluetoothAddress());
  println("Name: " + localDevice.getFriendlyName());




  val lock = new Object();

  val listener = new DiscoveryListener() {
    def deviceDiscovered(btDevice:RemoteDevice , cod:DeviceClass) {
      System.out.println("Device " + btDevice.getBluetoothAddress + " found");
      remotes = remotes ++ List(btDevice)
      try {
        System.out.println("     name " + btDevice.getFriendlyName(false));
      } catch
      {
        case _ @ e =>
      }
    }
    def inquiryCompleted( discType:Int) {
      System.out.println("Device Inquiry completed!");
      synchronized {
        notifyAll();
      }
    }
    def serviceSearchCompleted(transID:Int, respCode:Int) {}
    def servicesDiscovered(transID:Int, servRecord:Array[ServiceRecord] ) {}
  };

   synchronized {
            val started = LocalDevice.getLocalDevice.getDiscoveryAgent.startInquiry(DiscoveryAgent.GIAC, listener);
            if (started) {
                println("wait for device inquiry to complete...");
                wait();
                println(remotes.size +  " device(s) found");
            }
        }


}

File: kevoree-core/org.kevoree.kcl/src/main/java/org/kevoree/kcl/Tester.java
package org.kevoree.kcl;

/**
 * Created with IntelliJ IDEA.
 * User: jed
 * Date: 30/05/13
 * Time: 16:14
 * To change this template use File | Settings | File Templates.
 */
public class Tester {



    public static void main(String argv[]) throws Exception{

        KevoreeJarClassLoader kcl = new KevoreeJarClassLoader();
        kcl.add("/home/jed/.m2/repository/org/slf4j/slf4j-api/1.7.5/slf4j-api-1.7.5.jar");

        kcl.loadClass("org.slf4j.LoggerFactory");

        System.out.println("ici") ;
        kcl.add("/home/jed/.m2/repository/org/slf4j/slf4j-api/1.6.5/slf4j-api-1.6.5.jar");


        kcl.loadClass("org.slf4j.LoggerFactory");

    }
}
